docs/limits.md

```diff
@@ -0,0 +1,258 @@
+This document describes some of the limitations of `v1` of the AEA framework and tradeoffs made in its design.
+
+### Rejected ideas:
+
+#### Handlers implemented as behaviours:
+
+Handlers can be considered a special cases of a "behaviour that listens for specific events to happen".
+
+One could implement `Handler` classes in terms of `Behaviours`, after having implemented the feature that behaviours can be activated after an event happens (e.g. receiving a message of a certain protocol).
```

Comment:
maybe a sentence on why this has been rejected?

- better to have a clear separation of concerns, and avoid purely reactive (handlers) and proactive (behaviours) components are conflated into one concept.
- it adds complexity to behaviour development.
- other...